src/utils.js

```diff
@@ -4,6 +4,44 @@ async function asyncForEach(array, callback) {
   }
 }
 
+function getRequiredBlockConfirmations(contract) {
+  return contract.methods.requiredBlockConfirmations().call()
+}
+
+function waitForBlockConfirmations(web3, event, requiredBlockConfirmations, blockNumberProvider) {
+  const transactionReceiptAsync = async function(event, resolve, reject) {
+    const latestBlockNumber = blockNumberProvider.getLatestBlockNumber()
+    const blockConfirmations = latestBlockNumber - event.blockNumber + 1
+    if (blockConfirmations >= requiredBlockConfirmations) {
+      try {
+        const receipt = await web3.eth.getTransactionReceipt(event.transactionHash)
+        if (receipt && receipt.blockNumber) {
```

Comment:
we should probably also check for `event.blockNumber` === `receipt.blockNumber`
to make sure that original tx still has the same blockNumber as we received it. if not, that means it was already forked and was included in different block so we have to recount our block confirmations